### PR TITLE
refactor: use AST parsing for complex export name extraction

### LINKED ISSUE #273: Duplicate import warning when linking to a package
### Environment

Node v18.16.0, Nuxt 3.7.1, Unimport 3.3.0

### Reproduction

In v3.3.0 both the `.d.ts` and `.mjs` files are being imported from a linked package, causing import warnings.

### Describe the bug

I have a nuxt module that registers a directory of composables for auto-import. It works fine when installed to `node_modules`, but as of `unimport@3.3.0` there is now a warning that occurs whens the module is linked.

`Duplicated imports "useLog", the one from "[path to package]/dist/runtime/composables/core/useLog.d.ts" has been ignored`

It looks like `scanFilesFromDir` should be ignoring the `.d.ts` files, but it doesn't work when the package is linked.

### Additional context

_No response_

### Logs

_No response_

### LINKED ISSUE #500: 5.7.0 made unplugin-auto-import error
### Environment

.

### Reproduction

.

### Describe the bug

in js file, one function has param name `options`, but auto import it:

Image

that's result with 5.7.0:

Image

and in `use-global-i18n.ts` no export name `options`:

Image

that's result with 5.6.0:

Image

### Additional context

_No response_

### Logs

```sh

```

### LINKED ISSUE #502: v5.7.0: arrow function parameters parsed as exported declarations
scanExports in v5.7.0 incorrectly picks up arrow function parameters as named exports.

This file (from @nuxt/ui):

```js
export const useResizable = (key, options = {}, { collapsed = ref(false) } = {}) => { ... };
```

...causes `options` to be registered as an export alongside `useResizable`. At runtime:

```
SyntaxError: Importing binding name 'options' is not found.
```

The new comma-separated declaration parsing doesn't distinguish between:

```js
export const a = 1, b = 2, c = 3          // a, b, c are all exports
export const fn = (a, b = {}) => { ... }   // only fn is an export
```

Works on 5.6.0, broken on 5.7.0.

### LINKED ISSUE #34407: Issue with Unimport 5.7.0
I used the latest version of Nuxt (4.3.1) with Nuxt-UI (pro 1.8)...
Yesterday, I couldn't create a production version because Nuxt requires Unimport ^5.6.0, but npm install the latest version 5.7.0, which causes problems (see https://github.com/unjs/unimport/issues/500).

In my case, it's a serious problem, impossible to fix. I have hundreds of errors...

Result:
```
"options" is not exported by "node_modules/@nuxt/ui/dist/runtime/composables/defineShortcuts.js", imported by "app/components/Settings/Application.vue".
│  file: /Users/robertjuzyna/Dev/Drop pix/Unit-AI/dashboard-ce-matin/app/components/Settings/Application.vue:94:9
│
│  92: import { value } from '/... /node_modules/@nuxt/ui-pro/composabl...
│  93: import { useModal } from '/... /node_modules/@nuxt/ui/dist/runti...
│  94: import { options } from '/.../node_modules/@nuxt/ui/dist/runtim...
│               ^
│  95: const _sfc_main = {
│  96:   __name: "SettingsApplication",
│
```

Operating system      macOS 25.3.0                                                                                          │
│ │  CPU                   Apple M1 Pro (10 cores)                                                                               │
│ │  Node.js version       v22.18.0                                                                                              │
│ │  nuxt/cli version      3.33.1                                                                                                │
│ │  Package manager       npm 10.9.4                                                                                            │
│ │  Nuxt version          4.3.1                                                                                                 │
│ │  Nitro version         2.13.1                                                                                                │
│ │  Builder               vite 7.3.1                                                                                            │
│ │  Config                app, auth, compatibilityDate, css, devServer, devtools, experimental, extends, hooks, i18n, icon,     │
│ │                        mdc, modules, nitro, plugin, routeRules, runtimeConfig, site, ssr, tiptap, typescript, uiPro, ui,     │
│ │                        vite                                                                                                  │
│ │  Modules               @nuxt/ui 2.22.3, @nuxtjs/mdc 0.20.1, radix-vue/nuxt 1.9.17, @nuxt/icon 1.15.0, @nuxtjs/i18n 10.2.3,   │
│ │                        @vueuse/nuxt 13.9.0, @pinia/nuxt 0.11.3, pinia-plugin-persistedstate/nuxt 4.7.1, nuxt-auth-utils      │
│ │                        0.5.29, nuxt-rating 0.1.6, nuxt-tiptap-editor 3.2.0                                                   │
│ │

### PR DESCRIPTION
**Pull request by cernymatej**
fix https://github.com/unjs/unimport/issues/502
fix https://github.com/unjs/unimport/issues/500
fix https://github.com/nuxt/nuxt/issues/34432
fix https://github.com/nuxt/nuxt/issues/34407
fix https://github.com/unjs/unimport/issues/273 (possibly?)

This replaces @danielroe's solution for handling commas in export statemens from https://github.com/unjs/mlly/pull/336 with a proper AST-based parser, which should handle the remaining edge cases.

Probably the main difference compared the current implementation in `main` is that the function can now fail if we try to parse non-valid JavaScript syntax.

Other than that, it is _mostly_ identical in terms of performance in the majority of cases. We still use a regex to find the exports and only parse the ones that actually need it, reducing the performance impact significantly.

In the more complex cases where parsing does need to happen, all code slices are batched into a single `parseSync` call, which has also proven to speed things up a lot. The calls where we parse the code are ~1.5x slower than the old regex-only approach, but it no longer produces incorrect results.

----
For now, I haven't touched the part of the code that uses `acorn` for getting token locations. I haven't measured the performance of `acorn`, but I feel like tokenizing is much lighter than parsing the entire source file via `oxc-parser`... Maybe we can revisit this in the future?

## Summary by CodeRabbit

* **New Features**
  * More robust export name detection using AST-based batching for complex patterns (destructuring with renames/defaults, generators, multi-typed declarations, ES2022 string-literal names) and improved TypeScript/TSX handling.

* **Tests**
  * Expanded coverage for TSX/JSX, advanced destructuring, string-literal exports/types, generators, nested edge cases, and other export parsing scenarios.

### CODE REVIEW

File: src/analyze.ts [hunk 1]
```diff
@@ -511,16 +417,43 @@ export function findExports(code: string): ESMExport[] {
...
+  for (const destructuredExport of destructuredExports) {
+    needsParse.push(destructuredExport);
+  }
+
+  if (needsParse.length > 0) {
+    // Get the export names for each export
+    const parsedExports = _batchParseExportNames(needsParse.map((e) => e.code));
+    for (const [i, target] of needsParse.entries()) {
+      target.names = parsedExports[i];
+    }
```
[the remark this answers is not in the document]
Comment (cernymatej) [reply src/analyze.ts:456]: `oxc-parser` handles the error gracefully... can we just rely on it?

```js
Errors: [
  {
    "severity": "Error",
    "message": "Expected `)` but found `EOF`",
    "labels": [
      {
        "message": "`)` expected",
        "start": 26,
        "end": 26
      }
    ],
    "helpMessage": null,
    "codeframe": "\n  x Expected `)` but found `EOF`\n   ,-[input.ts:1:27]\n 1 | export const foo = fn(a, b\n   `----\n"
  }
]
     Static exports: [
  {
    "start": 0,
    "end": 26,
    "entries": [
      {
        "start": 7,
        "end": 26,
        "moduleRequest": null,
        "importName": {
          "kind": "None",
          "name": null,
          "start": null,
          "end": null
        },
        "exportName": {
          "kind": "Name",
          "name": "foo",
          "start": 13,
          "end": 16
        },
        "localName": {
          "kind": "Name",
          "name": "foo",
          "start": 13,
          "end": 16
        },
        "isType": false
      }
    ]
  }
]
```

File: test/exports.test.ts [hunk 2]
```diff
@@ -169,6 +187,30 @@ describe("findExports", () => {
...
+    expect(matches).toHaveLength(1);
+    expect(matches[0].names).toEqual(["foo", "bar"]);
+  });
+  it.fails("export const { foo, bar: { baz, qux } } = obj", () => {
+    const matches = findExports(
+      "export const { foo, bar: { baz, qux } } = obj",
+    );
+    expect(matches).toHaveLength(1);
+    expect(matches[0].names).toEqual(["foo", "baz", "qux"]);
+  });
```
Comment (cernymatej) [inline test/exports.test.ts:212]: btw. this didn't work before either

I'm not sure if we can support this case with only simple regex targeting to get the code slices for the parser. And parsing the whole file turned out to slow things down significantly.

**Comment by MindfulLearner** [comment]
This AST-based approach would also fix a case #336 misses: generics on the arrow function itself, not a type annotation before `=`.

```ts
export const useForm = <From, To>(schema: From, options?: To) => make(schema, options)
```

`findExports` still returns `names: ["useForm", "To"]` here, since `_extractExtraNames` only tracks angle brackets while `inTypeAnnotation` is true (after `:`, before `=`), not for generics right after `=`. In our case (Nuxt + `unimport`) this generated a bogus `To` re-export, causing a white screen at runtime. Worked around locally with a function declaration, happy to add a test case if useful for this!
